guillotina/files/field.py

```diff
@@ -145,11 +147,18 @@ def extension(self, val):
                     setattr(file_ob, serialize_mappings[key], item_value)
             data_context._p_register()
             return file_ob
+        value['data'] = base64.b64decode(value['data'])
     else:
+        # base64 web value
         value = convert_base64_to_binary(value)
-        request = get_current_request()
-        file_manager = get_multi_adapter((context, request, field), IFileManager)
-        val = await file_manager.save_file(
-            partial(_generator, value), content_type=value['content_type'],
-            size=len(value['data']))
-        return val
+
+    # There is not file and expecting a dict
+    # 'data', 'encoding', 'content-type', 'filename'
+    request = get_current_request()
+    file_manager = get_multi_adapter((context, request, field), IFileManager)
+    content_type = value['content_type'] if 'content_type' in value else value['content-type']
```

Comment:
This looks wrong.

Can we just use `value.get('content-type')`?

Reply:
The problem is that we used to have content_type as the key and plone rest api uses content-type